ocaml/typing/printtyp.ml

```diff
@@ -889,7 +898,7 @@ let rec get_best_path r env =
           match !r with
             Best p' when path_size p env >= path_size p' env -> ()
           | _ -> r := Best p)
-        l;
+        (List.rev l);
```

Comment:
It's essentially arbitrary but allows us to pick `Sexplib0.Sexp.t` for the sexp type, rather than `Parsexp.Single.parsed_value`. That's the main motiviating example.

Reversing also makes the output more stable. If two paths, `p1` and `p2`, score equally well as names for `p`, and `p1` is discovered first, then:
  * the old code would print `p1` until `p2` is discovered, then start printing `p2`
  * the new code will always print `p1`

This comes up in mdx tests.